src/Validator.php

```diff
@@ -131,6 +131,10 @@ public function validate(
                 $validatedData = $dataSet->getAttributeValue($attribute);
                 $context->setParameter(ValidationContext::PARAMETER_VALUE_AS_ARRAY, null);
                 $context->setAttribute($attribute);
+
+                if ($dataSet instanceof LabelsProviderInterface) {
+                    $context->setAttributeLabel($dataSet->getValidationPropertyLabels()[$attribute] ?? null);
+                }
```

Comment:
Seems, currently may be case when label will be incorrect (keep from previous context). 

I think, label setting should be here:

![image](https://github.com/yiisoft/validator/assets/525501/799ac227-d9ed-4e41-8b17-78720471b022)